src/caffe/test/test_filler.cpp

```diff
@@ -64,6 +65,51 @@ TYPED_TEST(UniformFillerTest, TestFill) {
 }
 
 template <typename Dtype>
+class UniformStaticFillerTest : public ::testing::Test {
+ protected:
+  UniformStaticFillerTest()
+      : blob_(new Blob<Dtype>(2, 3, 4, 5)),
+        filler_param_() {
+    filler_param_.set_min(1.);
+    filler_param_.set_max(2.);
+    filler_.reset(new UniformStaticFiller<Dtype>(filler_param_));
+    filler_->Fill(blob_);
+  }
+  virtual ~UniformStaticFillerTest() { delete blob_; }
+  Blob<Dtype>* const blob_;
+  FillerParameter filler_param_;
```

Comment:
Naked pointer for the Blob but `shared_ptr` for the filler? Why? Please use `shared_ptr` everywhere.